File: src/languages/java/binary/gates/gates.java
package binary.gates;

public class gates {
    
    public static boolean Not(boolean a) {
        return !a;
    }
    
    public static boolean And(boolean a, boolean b) {
        return a == b;
    }
    
    public static boolean Nand(boolean a, boolean b) {
        return Not(And(a, b));
    }
    
    public static boolean Nor(boolean a, boolean b) {
        return And(Not(a), Not(b));
    }
    
    public static boolean Or(boolean a, boolean b) {
        return Not(And(Not(a), Not(b)));
    }
    
    public static boolean Xor(boolean a, boolean b) {
        return Or(And(Not(a), b), And(a, Not(b)));
    }
    
    public static boolean Xnor(boolean a, boolean b) {
        return Not(Or(And(Not(a), b), And(a, Not(b))));
    }
    

}

File: src/languages/go/binary/gates/gates.go
package gates

func Not(a bool) bool {
	o := true
	if a == true {
		o = false
	}
	return o
}

func And(a, b bool) bool {
	o := false
	if a == true && b == true {
		o = true
	}
	return o
}

func Nand(a, b bool) bool {
	return Not(And(a, b))
}

func Nor(a, b bool) bool {
	return And(Not(a), Not(b))
}

func Or(a, b bool) bool {
	return Not(And(Not(a), Not(b)))
}

func Xor(a, b bool) bool {
	return Or(And(Not(a), b), And(a, Not(b)))
}

func Xnor(a, b bool) bool {
	return Not(Or(And(Not(a), b), And(a, Not(b))))
}
